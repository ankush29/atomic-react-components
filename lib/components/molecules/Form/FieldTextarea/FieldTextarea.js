// @flow
import React from 'react';
import type { Node } from 'react';
import { Field, ErrorMessage } from 'formik';
import styled from 'styled-components';
import Textarea from '../../../atoms/Textarea';
import FieldError from '../../../atoms/FieldError';
import Label from '../../../atoms/Label';
import styles from './FieldTextarea.style';

type Props = {
  id: string,
  className?: string,
  name: string,
  label: Node,
  labelProps?: {
    className?: string,
  },
  value?: string,
  onChange?: string,
};

const FieldTextarea = ({
  id,
  className,
  name,
  onChange,
  label,
  labelProps,
  ...others
}: Props): Node => (
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>
      {label}
    </Label>
    <Field id={id} name={name}>
      {({ field, form }) => {
        const { handleChange } = form;
        return (
          <Textarea
            {...others}
            {...field}
            onChange={e => {
              handleChange(e);
              if (typeof onChange === 'function') onChange(e, form);
            }}
          />
        );
      }}
    </Field>
    <ErrorMessage name={name} component={FieldError} />
  </div>
);

FieldTextarea.defaultProps = {
  className: '',
  labelProps: {},
  value: '',
  onChange: null,
};

export default styled(FieldTextarea)`
  ${styles};
`;

export { FieldTextarea as FieldTextareaVanilla };
