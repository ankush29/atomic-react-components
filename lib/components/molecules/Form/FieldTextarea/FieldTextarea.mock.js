const defaultTextareaForm = {
  initialValues: {
    comment: '',
  },
  action: '/',
  validationSchema: {
    comment: {
      max: [100, 'Max limit reached'],
      required: 'Required field',
    },
  },
  handleSubmit: () => {},
};

export default defaultTextareaForm;
