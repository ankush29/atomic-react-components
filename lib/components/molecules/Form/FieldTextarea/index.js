// @flow

export { default, FieldTextareaVanilla } from './FieldTextarea';
