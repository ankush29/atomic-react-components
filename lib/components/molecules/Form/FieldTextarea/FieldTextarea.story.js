// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import FieldTextarea from '.';
import Form from '../Form';
import defaultTextareaForm from './FieldTextarea.mock';

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module)
  .addParameters({ jest: ['FieldTextarea', 'Form'] })
  .addWithChapters('FieldTextarea', {
    chapters: [
      {
        sections: [
          {
            sectionFn: () => (
              <Form {...defaultTextareaForm}>
                <FieldTextarea
                  id="comment"
                  name="comment"
                  label="Comment"
                  placeholder="Enter your text here..."
                />
              </Form>
            ),
            options: {
              showSource: true,
              allowSourceToggling: true,
              showPropTables: true,
              allowPropTablesToggling: true,
            },
          },
        ],
      },
    ],
  });
