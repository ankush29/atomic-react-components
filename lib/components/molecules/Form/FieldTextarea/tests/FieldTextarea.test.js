import React from 'react';
import { shallowWithTheme, renderWithTheme } from '../../../../../../__mock__/themeMock';
import { FieldTextareaVanilla } from '../FieldTextarea';

describe('FieldTextarea Component', () => {
  let FieldTextarea = '';

  test.only('should render correctly', () => {
    FieldTextarea = shallowWithTheme(<FieldTextareaVanilla />);
    expect(FieldTextarea).toMatchSnapshot();
  });

  test('should render label correctly', () => {
    const props = {
      label: 'Comment',
    };
    FieldTextarea = renderWithTheme(<FieldTextareaVanilla {...props} />);
    expect(FieldTextarea.find('label').text()).toEqual(props.label);
  });

  test('should render textarea tag', () => {
    FieldTextarea = renderWithTheme(<FieldTextareaVanilla />);
    expect(FieldTextarea.find('textarea')).toHaveLength(1);
  });
});
