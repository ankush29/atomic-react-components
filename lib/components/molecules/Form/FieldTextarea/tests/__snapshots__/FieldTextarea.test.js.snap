// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldTextarea Component should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <FieldTextarea
    className=""
    errors={Object {}}
    labelProps={Object {}}
    onChange={null}
    register={[Function]}
    setValue={[Function]}
    value=""
  />
  <FieldTextarea
    className=""
    errors={Object {}}
    labelProps={Object {}}
    onChange={null}
    register={[Function]}
    setValue={[Function]}
    value=""
  />
  <Button
    ariaLabel=""
    className=""
    disabled={false}
    elementLocator=""
    inheritedStyles=""
    primary={true}
    secondary={false}
    tertiary={false}
    transparent={false}
    type="submit"
    variation=""
  >
    Submit
  </Button>
</form>
`;
